### PR TITLE
feat(frontend): add IC transactions to unified list util

### PR DESCRIPTION
**Pull request by AntonioVentilii**
# Motivation

Following PR #3547 , we add the IC transactions to the util that unifies the transactions.

# Tests

Provided additional tests.

### CODE REVIEW

File: src/frontend/src/lib/utils/transactions.utils.ts [hunk 1]
```diff
@@ -95,8 +106,24 @@ export const mapAllTransactionsUi = ({
 		}
 
 		if (isNetworkIdICP(networkId)) {
-			// TODO: Implement ICP transactions
-			return acc;
+			// TODO: Implement ckBTC and ckETH pending transactions
```
Comment (peterpeterparker) [inline src/frontend/src/lib/utils/transactions.utils.ts:109]: Unsolicited feedback: the more you provide PR for this feature, the more I think it's unrealisitc in terms of performance but, let's see.

File: src/frontend/src/tests/mocks/ic-transactions.mock.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import type { IcTransactionUi } from '$icp/types/ic-transaction';
+
+export const createMockIcTransactionsUi = (n: number): IcTransactionUi[] =>
+	Array.from({ length: n }, () => ({
+		id: Math.random().toString(36).substring(7),
```
Comment (peterpeterparker) [inline src/frontend/src/tests/mocks/ic-transactions.mock.ts:5]: `crypto.randomUUID()` ?

**Review by peterpeterparker — APPROVED** [review]
LGTm, thx

File: src/frontend/src/lib/utils/transactions.utils.ts (hunk 1 above)
Comment (AntonioVentilii) [reply src/frontend/src/lib/utils/transactions.utils.ts:109]: yes... the final version is not exactly good; we should really start thinking about caching

File: src/frontend/src/tests/mocks/ic-transactions.mock.ts (hunk 2 above)
Comment (AntonioVentilii) [reply src/frontend/src/tests/mocks/ic-transactions.mock.ts]: done!

**Comment by AntonioVentilii** [comment]
[OISY-130]

**Pull request merged**